### PR TITLE
Fix `OnGiven` not being triggered properly

### PR DESCRIPTION
**Pull request by 4z0t**

### CODE REVIEW

File: lua/SimUtils.lua [hunk 1]
```diff
@@ -188,78 +188,78 @@ function TransferUnitsOwnership(units, toArmy, captured)
...
+            newUnit.DefaultBuildRate = defaultBuildRate
+            newUnit.TargetUpgradeBuildTime = upgradeBuildTimeComplete
 
             upgradeUnitCount = upgradeUnitCount + 1
-            upgradeUnits[upgradeUnitCount] =  unit
+            upgradeUnits[upgradeUnitCount] = newUnit
         end
 
-        unit.IsBeingTransferred = nil
+        newUnit.IsBeingTransferred = nil
```
Comment (Garanas) [inline lua/SimUtils.lua:259]: this is set on the old unit, it is already nil on the new unit

**Comment by Garanas** [comment]
May want to update your `deploy/fafdevelop` branch and re-do this pr, see the conflicts

File: lua/SimUtils.lua [hunk 2]
```diff
@@ -170,13 +170,13 @@ function TransferUnitsOwnership(units, toArmy, captured)
         unit.IsBeingTransferred = true
 
         -- changing owner
-        unit = ChangeUnitArmy(unit, toArmy)
+        local newUnit = ChangeUnitArmy(unit, toArmy)
         if not unit then
```
Comment (speed2CZ) [inline lua/SimUtils.lua:174]: shouldn't this check the newUnit?

**Comment by 4z0t** [comment]
I have no idea why `SetHealth` take instigator xd

**Comment by Garanas** [comment]
@4z0t / @speed2CZ are these changes good?

**Review by Hdt80bro — APPROVED** [review]

**Pull request merged**